imagery-cesium/plugins/entities-visualizer/base-entities-visualizer.ts

```diff
@@ -25,45 +30,213 @@ export abstract class BaseEntitiesVisualizer extends BaseImageryVisualizer {
-					show: true,
-					position: Cesium.Cartesian3.fromDegrees(
-						entity.featureJson.geometry.coordinates[0],
-						entity.featureJson.geometry.coordinates[1],
-						entity.featureJson.geometry.coordinates[2] ? entity.featureJson.geometry.coordinates[2] : 0
-					),
-					pixelOffset: Cesium.Cartesian2.ZERO,
-					eyeOffset: Cesium.Cartesian3.ZERO,
-					heightReference: Cesium.HeightReference.NONE,
-					horizontalOrigin: Cesium.HorizontalOrigin.CENTER,
-					verticalOrigin: Cesium.VerticalOrigin.CENTER,
-					scale: 1.0,
-					image: entity.icon,
-					imageSubRegion: undefined,
-					color: Cesium.Color.WHITE,
-					id: id,
-					rotation: 0.0,
-					alignedAxis: Cesium.Cartesian3.ZERO,
-					width: undefined,
-					height: undefined,
-					scaleByDistance: undefined,
-					translucencyByDistance: undefined,
-					pixelOffsetScaleByDistance: undefined,
-					sizeInMeters: false,
-					distanceDisplayCondition: undefined
-				};
-				const feature = this.billboardCollection.add(billboardEntity);
-				this.idToEntity.set(id, <any>{
-					originalEntity: entity,
-					feature: feature
-				});
+			const featureJson: Feature<any> = entity.featureJson;
+
+			switch (featureJson.geometry.type) {
+				case 'Point': {
+					this.getOrCreateDataSource("Point").then(ds => {
+						if (entity.icon) {
+							ds.entities.add(this.getBillbord(entity.id, featureJson.geometry, entity.icon));
+						} else {
+							ds.entities.add(this.getPoint(entity.id, featureJson.geometry, entity.style));
+						}
+					});
+					break;
+				}
+				case 'LineString': {
+					this.getOrCreateDataSource("LineString").then(ds => {
+						ds.entities.add(this.getPolyline(entity.id, featureJson.geometry, entity.style));
+					});
+					break;
+				}
+				case 'Polygon': {
+					this.getOrCreateDataSource("Polygon").then(ds => {
+						ds.entities.add(this.getPolygon(entity.id, featureJson.geometry, entity.style));
+					});
+					break;
+				}
 			}
 		});
 		return of(true);
 	}
 
+
+	private getBillbord(id: string, geometry: Point, imgUrl: string): any {
+		return {
+			id: id,
+			position: geoToCesium.coordinatesToCartesian(geometry.coordinates),
+			billboard: {
+				image: imgUrl
+			}
+		};
+	}
+
+	private getPoint(id: string, geometry: Point, stylesState?: Partial<IVisualizerStateStyle>): any {
+		const styles = merge({}, stylesState);
+		const s: IVisualizerStyle = merge({}, styles.initial);
+		const ptColor = this.getColor(s["marker-color"]);
+		return {
+			id: id,
+			position: geoToCesium.coordinatesToCartesian(geometry.coordinates),
+			point: {
+				color: ptColor
+			}
+		}
+	}
+
+	private getPolyline(id: string, geometry: LineString, stylesState?: Partial<IVisualizerStateStyle>): any {
+
+		// TODO: Support all polyline styles
+		const styles = merge({}, stylesState);
+		const s: IVisualizerStyle = merge({}, styles.initial);
+
+		const lineColor = this.getColor(s["stroke"]);
+		const lineWidth = s['stroke-width'];
+
+		return {
```

Comment:
add an appropriate type in the imagery-cesium model

Reply:
As we discussed it a private function and cesium type script make it complicated